src/bin/pgcopydb/env_utils.c

```diff
@@ -215,3 +217,83 @@ get_env_pgdata_or_exit(char *pgdata)
 			  "or from --pgdata");
 	exit(EXIT_CODE_BAD_ARGS);
 }
+
+
+/*
+ * get_env_using_parser checks if the environment variable exists and if it does
+ * it parses it and stores it in the target. If the environment variable is not
+ * set, the target is not modified. If the environment variable is set but the
+ * parsing fails, an error is logged and the errors counter is incremented.
+ */
+void
+get_env_using_parser(EnvParser *parser, int *errors)
+{
+	if (env_exists(parser->envname))
+	{
```

Comment:
If you implement that the other way around (early exit), then you can have one less level of nesting, which will ease reading and maintenance.